### DISCUSSION TITLE
Check ROMS for missed

### DISCUSSION THREAD

**Discussion by IvanCaregnato** [retronas/discussions#65] [Ideas]
Hi guys, any chance to get a script that, according to SMDB, could check missed roms for every specific system supported by RETRONAS?

**Comment by sairuk** [comment]
The romimport tools should leave reports for missing roms in the romimport folder after a scan. I added that functionality a while back but may not have documented it in the wiki now that I think about it.

If you are running the romimport tool install from a while ago you may need to reinstall it to get the updated script where missing reports were managed

**Comment by IvanCaregnato** [comment]
It's not correct that thing about ROMIMPORT. As you run it, it only gives you the amount of good set for what you are importing. If i missed only 2 roms over 1.000 and I imported them via ROMIMPORT, the result will be 2 on 1000 (0,2%). I'd like it gives me 1000/1000 (100%), as I have the complete in that moment.

**Reply by sairuk** [reply]
Ahhh i see what you are asking for now, we could look into that or the community can have a go at it.

I have never used SMDB (I have my own tools) so i have no idea how it's supposed to work scanning sets in place. I expect we could provide functionality in a script for what you are asking and still output missing to the same place.

You can also take a look at SabreTools which is available for installation and will handle a lot of input/outputs

**Reply by sairuk** [reply]
Looks like we would use the verify-pack tool for this, OK

**Reply by sairuk** [reply]
This tools only pushes missing to stdout and doesn't creating a missing files txt report so we'd have to handle that. 

Be nice if the tooling was consistent in its approach

**Reply by IvanCaregnato** [reply]
Any chance to get it as an option for ROMIMPORT? It would be nice, even running it manually via terminal for specific system

**Reply by sairuk** [reply]
I am debating exactly that right now, not really sure which way to go. I think it'll be an alt tool but we may add an option to `import & verify` so too speak.

I'm not exactly happy with the code in the tools used for romimport since its limiting me from doing some stuff so I'm debating if i rewrite it as well.

**Reply by IvanCaregnato** [reply]
no hurry at all, thank you very much for your support man!

**Reply by sairuk** [reply]
I'll add this to the Ideas page in a minute so its not lost in the discussion boards
